### PR TITLE
feat: in scene scenario layer editor

### PR DESCRIPTION
**Pull request by mrbusysky**
I have updated the code to now allow for the removal of background and so on to happen in the scene window. It updates the sprite where it is.

You would need to select an object then ctrl+right mouse click

Features
* Flip of image
* Auto Cropping
* remove background
* set background
* and more

### CODE REVIEW

File: package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        {
+            GameObject selectedLayer = Selection.activeGameObject;
+            SpriteRenderer spriteRenderer = selectedLayer.GetComponent<SpriteRenderer>();
+            if (spriteRenderer != null)
+            {
+                spriteRenderer.flipY = !spriteRenderer.flipY;
+            }
+        }
+
+        private static void RemoveBackground()
```
Comment (TheLLspectre) [inline package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs:94]: Could you comment this method ?

File: package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                if (importer != null)
+                {
+                    importer.isReadable = true;
+                    importer.textureCompression = TextureImporterCompression.Uncompressed;
+                    importer.SaveAndReimport();
+                }
+
+                // Convert texture to uncompressed format
+                Texture2D uncompressedTexture = new Texture2D(texture2D.width, texture2D.height, TextureFormat.RGBA32, false);
+                uncompressedTexture.SetPixels(texture2D.GetPixels());
```
Comment (TheLLspectre) [inline package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs:114]: What about if the sprite selected is not from Scenario ? Default sprite ?

File: package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                    CommonUtils.FetchTextureFromURL(imageUrl, texture =>
+                    {
+                        // Update the sprite directly in Unity
+                        CommonUtils.ReplaceSprite(spriteRenderer, texture);
+                    });
+                });
+            }
+        }
+
+        private static void SetAsBackground()
```
Comment (TheLLspectre) [inline package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs:135]: Could you comment this method ?

File: package/Editor/Plugins/CommonUtils.cs [hunk 4]
```diff
@@ -233,5 +233,17 @@ public static void ApplyPixelsPerUnit(string filePath)
                 Debug.LogError("There was an issue when applying the Pixels Per Unit parameter. please restart the editor.");
             }
         }
+
+        public static void ReplaceSprite(SpriteRenderer spriteRenderer, Texture2D newTexture)
```
Comment (TheLLspectre) [inline package/Editor/Plugins/CommonUtils.cs:237]: Could you comment this method ?

**Review by TheLLspectre — CHANGES_REQUESTED** [review]

File: package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+{
+    public class SceneContextMenuActions
+    {
+        [InitializeOnLoadMethod]
+        private static void Initialize()
+        {
+            SceneView.duringSceneGui += OnSceneGUI;
+        }
+
+        private static void OnSceneGUI(SceneView sceneView)
```
[the remark this answers is not in the document]
Comment (mrbusysky) [reply package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs:16]: The private static you mean?

File: package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs (hunk 2 above)
Comment (TheLLspectre) [reply package/Editor/SceneTools/SceneContextMenuActions/SceneContextMenuActions.cs:152]: image

Still have this error on default sprite from Unity

**Review by TheLLspectre — CHANGES_REQUESTED** [review]

**Comment by mrbusysky** [comment]
Should now be sorted.

**Pull request closed without merging**